backend/src/main/kotlin/io/github/projektalmanac/amoxtli/backend/service/UserServiceKt.kt

```diff
@@ -30,4 +29,26 @@ open class UserServiceKt {
 
         return ExchangeMapper.INSTANCE.toGetIntercambios200ResponseDto(intercambios)
     }
+
+    fun aceptarIntercambio(
+        idUsuario: Int?,
+        idIntercambio: Int?,
+        aceptarIntercambioRequestDto: AceptarIntercambioRequestDto?
+    ): IntercambioDto? {
+
+        val usuario = userRepository.findById(idUsuario).orElseThrow { UserNotFoundException(idUsuario) }
+
+        val intercambio = exchangeRepository.findByIdAndUserAccepting(idIntercambio, usuario).orElseThrow { IntercambioNotFoundException(idIntercambio)}
+
+        usuario.removeExchangesOfferor(intercambio)
+
+        intercambio.bookOfferor = bookRepository.findById(aceptarIntercambioRequestDto?.idLibro?.toInt() ?: throw InvalidIdException()) ?: throw ResourceNotFoundException("El libro no existe") 
```

Comment:
```suggestion
        val idLibro = aceptarIntercambioRequestDto?.idLibro?.toInt() ?: throw InvalidIdException()
        intercambio.bookOfferor = bookRepository.findById(idLibro) ?: throw ResourceNotFoundException("El libro no existe") 
```

No pongas tantas cosas en una sola línea. Casi me da algo leyendo el código